### PR TITLE
Add captureProfile method to hub and client

### PR DESCRIPTION
**Pull request by stefanosiano**
## :scroll: Description
I added the method `captureProfile` to `IHub` and `ISentryClient`, so the profiles can be sent in separate envelopes instead of being attached to transactions.
Since the methods that send transaction and profile together were already exposed e.g. in `HubAdapter` or `SentryClient`, they were deprecated to avoid breaking changes.

## :bulb: Motivation and Context
We want to separate Profiling and Performance. 
The first step to do so is to send separate envelopes for them.
This will also allows to send timed out profiles without waiting for a transaction to be attached to (more on this on another pr)

## :green_heart: How did you test it?
I added few unit tests for the new methods and added one to check the deprecated method calls the new ones.

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

## :crystal_ball: Next steps
The profiler will send the profile envelope by itself, instead of relying on `SentryTracer`, but will be done in another pr

### CODE REVIEW

File: sentry/src/test/java/io/sentry/SentryClientTest.kt [hunk 1]
```diff
@@ -1038,39 +1039,55 @@ class SentryClientTest {
...
     @Test
     fun `when captureTransaction with attachments and profiling data`() {
         val transaction = SentryTransaction(fixture.sentryTracer)
-        fixture.getSut().captureTransaction(transaction, null, createScopeWithAttachments(), null, fixture.profilingTraceData)
-        verifyAttachmentsInEnvelope(transaction.eventId)
-        verifyProfilingTraceInEnvelope(transaction.eventId)
+        val client = fixture.getSut()
+        client.captureTransaction(transaction, null, createScopeWithAttachments(), null)
+        client.captureProfile(fixture.profilingTraceData)
+        verifyProfilingTraceInEnvelope(SentryId(fixture.profilingTraceData.profileId))
```
Comment (philipphofmann) [inline sentry/src/test/java/io/sentry/SentryClientTest.kt:1051]: `m`: I think we should split that test case up into two test cases. Previously, we had one call on the client. Now we have two and only verify for the profiling trace in the envelope. 

Something like
```Kotlin
client.captureTransaction(transaction, null, createScopeWithAttachments(), null)
verifyAttachmentsInEnvelope(transaction.eventId)
```
and
```Kotlin
client.captureProfile(fixture.profilingTraceData)
verifyProfilingTraceInEnvelope(SentryId(fixture.profilingTraceData.profileId))
```

That also applies to some of the other test cases.

File: sentry/src/test/java/io/sentry/HubTest.kt [hunk 2]
```diff
@@ -1312,6 +1281,61 @@ class HubTest {
...
+        val contexts = TransactionContext("name", "op", TracesSamplingDecision(false, null, true, null))
+        val transaction = hub.startTransaction(contexts)
+        transaction.finish()
+        verify(mockClient, never()).captureProfile(any())
+
+        // Transaction is sampled, so it should be profiled
+        val sampledContexts = TransactionContext("name", "op", TracesSamplingDecision(true, null, true, null))
+        val sampledTransaction = hub.startTransaction(sampledContexts)
+        sampledTransaction.finish()
+        verify(mockClient, never()).captureProfile(any())
```
Comment (philipphofmann) [inline sentry/src/test/java/io/sentry/HubTest.kt:1320]: `m`: Shouldn't this be
```suggestion
        verify(mockClient).captureProfile(any())
```

**Review by philipphofmann — APPROVED** [review]
Just some comments about the tests, apart from that LGTM

File: sentry/src/test/java/io/sentry/SentryClientTest.kt (hunk 1 above)
Comment (stefanosiano) [reply sentry/src/test/java/io/sentry/SentryClientTest.kt]: yep. Actually, there were already those tests, so i just removed the transactions from the captureProfile tests

**Pull request merged**